pupil_src/shared_modules/marker_tracker_3d/optimization/model_optimizer.py

```diff
@@ -0,0 +1,75 @@
+import background_helper
+from marker_tracker_3d import utils
+from marker_tracker_3d.model_optimizer_storage import ModelOptimizerStorage
+from marker_tracker_3d.optimization.optimization_generator import optimization_generator
+from marker_tracker_3d.optimization.visibility_graphs import VisibilityGraphs
+
+
+class ModelOptimizer:
+    def __init__(self, camera_model, update_menu=None):
+        self.camera_model = camera_model
+
+        self.model_optimizer_storage = ModelOptimizerStorage()
+        self.origin_marker_id = None
+        self.visibility_graphs = VisibilityGraphs(
+            self.model_optimizer_storage,
+            self.camera_model,
+            self.origin_marker_id,
+            update_menu,
+        )
+
+        self.bg_task = None
+
+    def update(self, marker_detections, camera_extrinsics):
+        self.visibility_graphs.add_marker_detections(
+            marker_detections, camera_extrinsics
+        )
+
+        self._run_optimization()
+
+        return self._get_updated_3d_marker_model()
+
+    def _run_optimization(self):
+        if not self.bg_task:
```

Comment:
Is the logic correct here? After you have created `bg_task` for the first time, it will as far as I can see never be `None` again. How is the second optimization started?